src/main/kotlin/com/sovdee/skriptbounds/elements/Types.kt

```diff
@@ -0,0 +1,115 @@
+                        "Represents a bounding box that's aligned along the global x, y, and z axis."
+                    )
+                    .serializer(object : Serializer<AxisAlignedBox>() {
+                        override fun serialize(o: AxisAlignedBox): Fields {
+                            val fields = Fields()
+                            fields.putPrimitive("minx", o.min.x)
+                            fields.putPrimitive("miny", o.min.y)
+                            fields.putPrimitive("minz", o.min.z)
+                            fields.putPrimitive("maxx", o.max.x)
+                            fields.putPrimitive("maxy", o.max.y)
+                            fields.putPrimitive("maxz", o.max.z)
+                            return fields
+                        }
+
+                        override fun canBeInstantiated() = false
+
+                        override fun mustSyncDeserialization() = false
+
+                        override fun deserialize(o: AxisAlignedBox?, f: Fields?) {
+                            throw IllegalStateException("AABs have no nullary constructor so this shouldn't be called.")
+                        }
+
+                        override fun deserialize(fields: Fields): AxisAlignedBox {
+                            val min = Vector3d(
+                                fields.getAndRemovePrimitive("minx", Double::class.java),
+                                fields.getAndRemovePrimitive("miny", Double::class.java),
+                                fields.getAndRemovePrimitive("minz", Double::class.java)
+                            )
+                            val max = Vector3d(
+                                fields.getAndRemovePrimitive("maxx", Double::class.java),
+                                fields.getAndRemovePrimitive("maxy", Double::class.java),
+                                fields.getAndRemovePrimitive("maxz", Double::class.java)
+                            )
+                            return AxisAlignedBox(min, max)
+                        }
+                    })
+            )
+
+            Classes.registerClass(
+                ClassInfo(Sphere::class.java, "sphericalboundingbox")
+                    .user("spher(e|ical) ?bounding ?box(e?s)?")
+                    .name("Spherical Bounding Box")
+                    .description(
+                        "Represents a bounding box that's a sphere centered on a point."
+                    )
+                    .serializer(object : Serializer<Sphere>() {
+                        override fun serialize(o: Sphere): Fields {
+                            val fields = Fields()
+                            fields.putPrimitive("centerx", o.center.x)
+                            fields.putPrimitive("centery", o.center.y)
+                            fields.putPrimitive("centerz", o.center.z)
+                            fields.putPrimitive("rsquared", o.radiusSquared)
+                            return fields
```

Comment:
Maybe:
```suggestion
                            Fields().apply {
                                putPrimitive("centerx", o.center.x)
                                putPrimitive("centery", o.center.y)
                                putPrimitive("centerz", o.center.z)
                                putPrimitive("rsquared", o.radiusSquared)
                            }
```
I would suggest `with` syntax but it only returns the explicit final value of the block (so you would put `this` at the end of the block).
```kt
with (Fields()) {
    ...
    this
}